ddl/ddl_api.go

```diff
@@ -4425,14 +4435,25 @@ func (d *ddl) CreateIndex(ctx sessionctx.Context, ti ast.Ident, keyType ast.Inde
 	// to job queue, the fail path logic is super fast.
 	// After DDL job is put to the queue, and if the check fail, TiDB will run the DDL cancel logic.
 	// The recover step causes DDL wait a few seconds, makes the unit test painfully slow.
+	// For same reason, decide whether index is global here.
 	indexColumns, err := buildIndexColumns(append(tblInfo.Columns, hiddenCols...), indexPartSpecifications)
 	if err != nil {
 		return errors.Trace(err)
 	}
+
+	global := false
 	if unique && tblInfo.GetPartitionInfo() != nil {
```

Comment:
tiny requirement question:

do we need support non-unique global index?

cc: @tiancaiamao  @zz-jason @coocood

Reply:
maybe it's useful to support "using a non-unique index without include partition columns", maybe there is a real-world example:

```
create table order (
  order_id int,
  user_id  int,
  seller_id int,
  primary key(order_id, user_id),
  index uid (user_id),
) partition by hash(user_id) partition 1000
```
this table will conventional in the "user end side" to let the order-system query by `order_id` and `user_id`.

but the seller also want query the order belongs itself in seller system and delivery order, and query by `seller_id` and it didn't know partition_id(also seller_id isn't unique in order table), and maybe we need a global non-unique index --- `index sid (seller_id)` to help seller request can be prunning?

Reply:
I agree. In some case global non-unique index is useful. Maybe we should support keywords `Local` and `Global` just like Oracle.  @zhaox1n had submitted a [PR](https://github.com/pingcap/parser/pull/915) about it, but it is closed now.

Reply:
Not in this PR.
Support global unique index is complex enough, we should do it step by step.